tss-esapi/tests/integration_tests/structures_tests/pcr_tests/pcr_selection_tests.rs

```diff
@@ -69,39 +70,243 @@ fn test_size_of_select_handling() {
+        PcrSelectSize::ThreeOctets,
+        &[PcrSlot::Slot5],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_1");
+
+    let pcr_selection_2 = PcrSelection::create(
+        HashingAlgorithm::Sha256,
+        PcrSelectSize::TwoOctets,
+        &[PcrSlot::Slot6],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_2");
+
+    assert_eq!(
+            Err(Error::WrapperError(WrapperErrorKind::InvalidParam)),
+            pcr_selection_1.merge_exact(&pcr_selection_2),
+            "Merge exact PcrSelections with different size of select did not produce the expected error",
+        );
+}
+
+#[test]
+fn test_merge_exact_non_unique_pcr_slot_errors() {
+    let mut pcr_selection_1 = PcrSelection::create(
+        HashingAlgorithm::Sha256,
+        PcrSelectSize::ThreeOctets,
+        &[PcrSlot::Slot5],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_1");
+
+    let pcr_selection_2 = PcrSelection::create(
+        HashingAlgorithm::Sha256,
+        PcrSelectSize::ThreeOctets,
+        &[PcrSlot::Slot5],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_2");
+
+    assert_eq!(
+        Err(Error::WrapperError(WrapperErrorKind::InvalidParam)),
+        pcr_selection_1.merge_exact(&pcr_selection_2),
+        "Merge exact PcrSelections with non unique PcrSlot did not produce the expected error",
+    );
+}
+
+#[test]
+fn test_subtract_exact_hasahing_algorithm_mismatch_errors() {
+    let mut pcr_selection_1 = PcrSelection::create(
+        HashingAlgorithm::Sha256,
+        PcrSelectSize::TwoOctets,
+        &[PcrSlot::Slot5],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_1");
+
+    let pcr_selection_2 = PcrSelection::create(
+        HashingAlgorithm::Sha384,
+        PcrSelectSize::TwoOctets,
+        &[PcrSlot::Slot6],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_2");
+
+    assert_eq!(
+            Err(Error::WrapperError(WrapperErrorKind::InconsistentParams)),
+            pcr_selection_1.subtract_exact(&pcr_selection_2),
+            "Subtract exact PcrSelections with different hashing algorithm did not produce the expected error",
+        );
+}
+
+#[test]
+fn test_subtract_exact_size_of_select_mismatch_errors() {
+    let mut pcr_selection_1 = PcrSelection::create(
+        HashingAlgorithm::Sha256,
+        PcrSelectSize::ThreeOctets,
+        &[PcrSlot::Slot5],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_1");
+
+    let pcr_selection_2 = PcrSelection::create(
+        HashingAlgorithm::Sha256,
+        PcrSelectSize::TwoOctets,
+        &[PcrSlot::Slot6],
+    )
+    .expect("Failed to create PcrSelection pcr_selection_2");
+
+    assert_eq!(
+            Err(Error::WrapperError(WrapperErrorKind::InvalidParam)),
+            pcr_selection_1.subtract_exact(&pcr_selection_2),
+            "Subtract exact PcrSelections with different size of select did not produce the expected error",
+        );
```

Comment:
For both this test and the next one - if you change the stuff that's currently causing the error, would it still error? I'm thinking here of two things: would it cause an error because the 2nd selection contains a slot that the first doesn't? And, would it cause an error if you end up with an empty selection at the end?

Reply:
The order in which things are checked is HashingAlgorithm->PcrSelectSize->slots. So I tried to only change one parameter at the time in order to trigger the errors.

Reply:
Oh I see what you mean. That is not correct.

Reply:
Fixed